main_loop_tf/main.py

```diff
@@ -676,6 +795,10 @@ def main_loop(placeholders, val_placeholders, train_outs, train_summary_op,
     last_epoch = False
     history_acc = np.array([]).tolist()
 
+    # Compute smaller batch if any
+    smaller_batch_size = (train.nsamples -
+                          (train.nbatches - 1) * dataset_params['batch_size'])
```

Comment:
You don't need this. Just check for batch_size rather than for smaller_batch_size